### PR TITLE
Redesigned user management

### PR DESCRIPTION
**Pull request by kdmnk**
Also made a page for mass handling statuses for collegists.

### CODE REVIEW

**Comment by kdmnk** [comment]
And do we have policies for status handling? Do we need, anyways? (I mean we might use the permission policies.)

**Comment by kdmnk** [comment]
Ooops, I forgot to add the workshop tags.. Will do it tomorrow.

**Comment by kdmnk** [comment]
The target is to set extern/resident role only for currently active collegists.
I have added this to the semesterSeeder, but I can test that only if the seeders and the hasRoleBase() works properly.

File: database/seeders/SemesterSeeder.php [hunk 1]
```diff
@@ -25,13 +25,26 @@ public function run()
             $semester = $semester->succ();
         }
 
-        $users = Role::getUsers(Role::COLLEGIST);
```
Comment (luksan47) [inline database/seeders/SemesterSeeder.php:28]: This should be used.

File: app/Http/Controllers/UserController.php [hunk 2]
```diff
@@ -91,11 +92,32 @@ public function updatePassword(Request $request)
         return redirect()->back()->with('message', __('general.successful_modification'));
     }
 
+    public function setCollegistType(Request $request)
```
Comment (luksan47) [inline app/Http/Controllers/UserController.php:95]: This seems too complicated. What are you trying to do here?

**Review by luksan47 — CHANGES_REQUESTED** [review]
Only had a look at the permission related things to unblock you.

File: database/seeders/SemesterSeeder.php [hunk 3]
```diff
@@ -25,13 +25,26 @@ public function run()
...
+            return $value->hasRoleBase(Role::COLLEGIST);
+        });
 
-        $semesters = Semester::all();
+        $semesters = Semester::all()->unique();
         foreach ($semesters as $semester) {
             foreach ($users as $user) {
                 $status = array_rand(Semester::STATUSES);
                 $user->allSemesters()->attach($semester, ['status' => Semester::STATUSES[$status]]);
+                if ($semester->tag() == Semester::current()->tag()) {
```
Comment (luksan47) [inline database/seeders/SemesterSeeder.php:37]: This should be done in the User seeder, not here.

File: app/Http/Controllers/UserController.php (hunk 2 above)
Comment (kdmnk) [reply app/Http/Controllers/UserController.php:94]: This sets the collegist resident/extern status.
I simplified this with new functions in user model.

Comment (kdmnk) [reply app/Http/Controllers/UserController.php:94]: however, still not works :/

**Review by luksan47 — APPROVED** [review]

**Pull request merged**